### DISCUSSION TITLE
IPhone App sync failed

### DISCUSSION THREAD

**Discussion by Mosconi8** [dani-garcia/vaultwarden#3616] [Q&A]
Hi,

since today, my Iphone Bitwarden app is unable to sync to my vaultwarden server. "Sync failed" I tested the sync from an other Iphone (to the same vw-server) and everything works fine. Also from other clients everything is fine.  The Bitwarden app is the newest one. The timestamp is ok. 
The vw version is also the newest and the docker container was also restartet.

Any ideas what i can do? 

Regard...

**Comment by matthiasg** [comment]
Any update on this ? we have the same issues and could supply debug info if needed

**Comment by Masgalor** [comment]
I don't have any knowledge about IPhones but I learned that most connection problems (no matter what app or service) are a matter of the device not trusting the TLS-connection.

Apple isn't really backporting any security related features, so devices might have an outdated trust-store don't support modern ciphers or protocols. So maybe the first thing to check would be if you can open your webvault with safari without getting any security related errors.

For everything else we would need to see the logs from the app and/or the server, I think.

**Reply by Mosconi8** [reply]
Sorry, i have no update to this issue. 

@ Masgalor: It's no problem to open the webvault via safari on the iPhone. Only the app makes problems. But where and how can i find the log file from this App? From the server i think, it's not a problem.

**Comment by BlackDex** [comment]
Do you see any connection to Vaultwarden at all (Or the reverse proxy for that matter) when trying to login?
Have you tried to fully uninstall and re-install the app and set the correct self-hosted host and tried to login again?

Since Vaultwarden doesn't maintain the clients any debug info from the clients will not help.
And the clients for both iOS and Android basically use the code-base, and my Android seems to be fine with connecting.

As stated above, it could be a certificate issue in which the clients just deny it, but that should probably also cause issues with Safari on the phones.

**Reply by Mosconi8** [reply]
Hello BlackDex,

i don't think nginx is the problem, because all the other (bitwarden extensions, apps) work fine. Yes i fully uninstalled the app + the data and re-installed it. After that i used the self-hosted adress to login into the app. That works fine, also with 2fa. So i think nginix reverse proxy is not the problem.

**Reply by BlackDex** [reply]
So, what doesn't work then??

**Reply by BlackDex** [reply]
Or a better question, do you see any errors in the Vaultwarden logs?
If not, and all other clients work, it probably is either a vault item which causes an issue, or maybe you changed to Argon2 and have very high settings?

**Reply by Mosconi8** [reply]
When i login into the Bitwarden app my passwords don't appear. The tresor is empty.

**Reply by Mosconi8** [reply]
i can't find a log file. So i think i have to define a log file.

**Reply by BlackDex** [reply]
If using docker, try `docker logs -f vaultwarden` or whatever you named the vaultwarden container

**Comment by Mosconi8** [comment]
So i'm back....
thanks for waiting!

This is the log file.... 
Could this be the problem?
web_files) GET /<p..> [10] => 404 Not Found

Question: Is the sync only a oneway sync from the iPhone to the vw-server or should i also get the updated passwords from the server on my iPhone?


`/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                           Version 1.29.0                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://github.com/dani-garcia/vaultwarden/discussions or        |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] Using saved config from `data/config.json` for configuration.

[2023-07-16 08:32:32.373][vaultwarden::api::notifications][INFO] Starting WebSockets server on 0.0.0.0:3012
[2023-07-16 08:32:32.390][start][INFO] Rocket has launched from http://0.0.0.0:80
[2023-07-16 08:33:06.863][request][INFO] GET /icons/forum.proxmox.com/icon.png
[2023-07-16 08:33:06.863][request][INFO] GET /icons/10.0.1.13/icon.png
[2023-07-16 08:33:06.865][vaultwarden::api::icons][WARN] Unable to download icon: Host resolves to a non-global IP. 
[2023-07-16 08:33:06.865][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2023-07-16 08:33:06.866][response][INFO] (icon_internal) GET /icons/<domain>/icon.png => 200 OK
[2023-07-16 08:35:29.027][request][INFO] POST /identity/connect/token
[2023-07-16 08:35:29.028][request][INFO] POST /identity/connect/token
[2023-07-16 08:35:29.040][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-07-16 08:35:29.042][response][INFO] (login) POST /identity/connect/token => 200 OK
[2023-07-16 08:35:29.083][request][INFO] GET /api/config/
[2023-07-16 08:35:29.084][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:35:29.092][request][INFO] GET /api/config/
[2023-07-16 08:35:29.093][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:35:29.991][request][INFO] GET /api/sync
[2023-07-16 08:35:30.034][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-07-16 08:35:30.152][request][INFO] GET /api/config/
[2023-07-16 08:35:30.153][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:35:34.644][request][INFO] PUT /api/devices/identifier/56.......863/token
[2023-07-16 08:35:34.646][response][INFO] (put_device_token) PUT /api/devices/identifier/<uuid>/token => 200 OK
[2023-07-16 08:35:38.885][request][INFO] GET /notifications/hub?access_token=a.......W
[2023-07-16 08:35:38.886][response][INFO] (web_files) GET /<p..> [10] => 404 Not Found
[2023-07-16 08:36:07.537][request][INFO] GET /api/sync
[2023-07-16 08:36:07.569][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-07-16 08:36:07.665][request][INFO] GET /api/config/
[2023-07-16 08:36:07.665][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:36:25.363][request][INFO] GET /api/sync
[2023-07-16 08:36:25.390][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-07-16 08:36:25.508][request][INFO] GET /api/config/
[2023-07-16 08:36:25.509][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:36:29.029][request][INFO] GET /api/sync
[2023-07-16 08:36:29.050][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-07-16 08:36:29.212][request][INFO] GET /api/config/
[2023-07-16 08:36:29.212][response][INFO] (config) GET /api/config => 200 OK
[2023-07-16 08:36:32.228][request][INFO] GET /api/sync
[2023-07-16 08:36:32.257][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2023-07-16 08:36:32.424][request][INFO] GET /api/config/
[2023-07-16 08:36:32.425][response][INFO] (config) GET /api/config => 200 OK`

**Comment by BlackDex** [comment]
Or all looks fine. The sync endpoint only sends the current vault items in the database. Since that call ended in a 200, that all works fine.

Since you also mentioned it works fine on all other clients, i think it's either the KDF settings are to high or, as mentioned before there is a vault item causing issues. Large notes for example with a high KDF could cause some issues on some devices.

So, it doesn't look like a Vaultwarden server side issue we can fix.

**Reply by Mosconi8** [reply]
Thank you BlackDex. Today i saw that there was an update of the bitwarden app available. After installation and correct login I was not really happy with the result. The same problem...

**Reply by BlackDex** [reply]
And, if you revert to the previous version of Vaultwarden it works?
If not, then I still think it's one of the items mentioned in my previous post

**Reply by Mosconi8** [reply]
Sorry, do you mean a previous version of the Bitwarden app or Vaultwarden?
I looked for a solution to install a previous version of the bitwarden app. But therefor i need a old itunes version and this old itunes version was not able to handle the privious bitwarden app. The Bitwarden app worked fine with the old vaultwarden version But from one day to another the app didn't show me my passwords anymore. I know there was also an IOS update and a bitwarden update 2 days before.

**Comment by matthiasg** [comment]
How would a vault item not be an issue with the Server ?In our case most items are automatically generated in exactly the same way with same password rules.A kind of bisect for the data might help to pinpoint it if that can cause issues. Sent from my iPhoneOn Jul 16, 2023, at 11:02, Mathijs van Veluw ***@***.***> wrote:﻿
Or all looks fine. The sync endpoint only sends the current vault items in the database. Since that call ended in a 200, that all works fine.
Since you also mentioned it works fine on all other clients, i think it's either the KDF settings are to high or, as mentioned before there is a vault item causing issues. Large notes for example with a high KDF could cause some issues on some devices.
So, it doesn't look like a Vaultwarden server side issue we can fix.

**Reply by BlackDex** [reply]
Vaultwarden just stores the data received from the clients. And again, since all other clients do work, that is all fine i think.

Also i have no clue on how to debug the iPhone app, so i can't help there.

**Reply by matthiasg** [reply]
One way to scope down to the problematic items would be to bisect the data via some sort of runtime parameter. e.g only load first half of items then second half etc.

**Reply by BlackDex** [reply]
Loading just a half means removing the other part.
You can also look in the database at the dates when a cipher was updated, and check all ciphers after it started.

Or, at least create a new account and verify you can login there, and load x amount of vault items from the broken account over, and see when ot breaks.

I'm interested into what triggers it though. But I'm afraid we can't prevent issues, unless we send something broken.

**Reply by Mosconi8** [reply]
Thanx for this great idea. 
I think i will start with a new user and with one of the old passwords. Then there comes another old pw and so on.

**Reply by Mosconi8** [reply]
Hello all together,

a few minutes ago i installed a new user. Then i made a new secret on vaultwarden. After sync. the secret got on my Bitwarden app. So i think it has to be done with one of my passwords. And i noticed, that i have to read the documentation from Bitwarden about organisations and so on.
Thank you very much so far!!!!!

**Reply by Mosconi8** [reply]
One question to this topic:
right now i have 2 user:
"Max" = the owner taht has all rights.!
"testuser" = the new user for the secret test

"web vault max" = web vault login with owner max.
"web vault testuser" = web vault login with the testuser

While i have no permission with "testuser" to access to secrets from "Max" i thought i can move one secret from "Max" to "testuser". In "web vault Max" I put one secret in the the "testuser vault". But i cannot see this secret in the "web vault testuser" nor in my iphone app (logged in as testuser). The "testuser" vault in the "web vault max" seems to be an other folder, that has nothing to do with secret folder from "testuser". 
But how can i access the sync folder (from "testuser") with the owner "Max".
